### PR TITLE
Add view model data for the projects list endpoint

### PR DESCRIPTION
**Pull request by nakhbari**

### CODE REVIEW

File: app/features-json/project_json_controller.rb [hunk 1]
```diff
@@ -9,8 +10,11 @@ class ProjectJSONController < AuthenticatedControllerBase
       current_provider_credential = check_and_get_provider_credential
       current_user_config_service = self.current_user_config_service
       all_projects = current_user_config_service.projects(provider_credential: current_provider_credential)
+      all_projects_views_models = all_projects.map do |project|
+        ProjectSummaryViewModel.viewmodel_from!(object: project)
```
Comment (minuscorp) [inline app/features-json/project_json_controller.rb]: I think we can do this in a single line:

```ruby
.map(&ProjectSummaryViewModel.method(: viewmodel_from!))
```

File: app/features-json/view_models/view_model.rb [hunk 2]
```diff
@@ -67,7 +67,7 @@ def viewmodel_from!(object:)
         raise "Override base_model(model) in order to use the ViewModel mixin." if @base_model.nil?
```
Comment (minuscorp) [inline app/features-json/view_models/view_model.rb]: I think we should remove the named parameter `object:` and just keep it with `object`, that will clean things up and does not add any useful information. What do you think?

File: app/features-json/view_models/view_model.rb [hunk 3]
```diff
@@ -67,7 +67,7 @@ def viewmodel_from!(object:)
...
         raise "Incorrect object type. Expected #{@base_model}, got #{object.class}" unless object.kind_of?(@base_model)
         if object.respond_to?(:to_object_dictionary)
-          return object.to_object_dictionary(ignore_instance_variables: @base_model.instance_variables - included_attributes)
+          return object.to_object_dictionary(ignore_instance_variables: object.instance_variables - included_attributes)
```
[the remark this answers is not in the document]
Comment (nakhbari) [reply app/features-json/view_models/view_model.rb]: The @base_model had an empty list of instance variables. Is this to be expected?

File: app/features-json/project_json_controller.rb (hunk 1 above)
Comment (nakhbari) [reply app/features-json/project_json_controller.rb]: nice!

File: app/features-json/view_models/view_model.rb (hunk 2 above)
Comment (nakhbari) [reply app/features-json/view_models/view_model.rb]: Not sure i'm following what you mean, can you propose something in a follow up PR?

File: app/features-json/view_models/view_model.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (minuscorp) [reply app/features-json/view_models/view_model.rb]: You're right, totally expected, `@base_model` is a class, not an instance.

**Comment by nakhbari** [comment]
@minuscorp just read a little of the ruby style guide. It seems that we should only use the `!` at the end of a method if there is a safe version of that dangerous method.

https://github.com/bbatsov/ruby-style-guide#dangerous-method-bang

So i'm gonna remove than bang for now.

**Comment by minuscorp** [comment]
Well, in the backend we are using the bang for methods that potentially raise exceptions, I think we should keep the bang just for clarity that the method is not safe if used wrongly

**Comment by nakhbari** [comment]
re-added the bang

File: app/features-json/view_models/view_model.rb (hunk 2 above)
Comment (taquitos) [reply app/features-json/view_models/view_model.rb]: @minuscorp it looks like that's already done 🎉 is there anything else that's missing from this PR?

Comment (minuscorp) [reply app/features-json/view_models/view_model.rb]: I think this: https://github.com/fastlane/ci/pull/566 supersedes this, but @nakhbari should decide 😄

**Pull request closed without merging**